vllm/multimodal/image.py

```diff
@@ -42,72 +40,41 @@ def get_dummy_image_data(
     """Standard dummy data factory for image data (to be used in
     :meth:`vlm.multimodal.MultiModalRegistry.register_dummy_data`)."""
     seq_data = _get_dummy_seq_data(seq_len, vlm_config)
-    values = _get_dummy_values(vlm_config)
+    image = _get_dummy_image(vlm_config)
 
-    config_input_type = vlm_config.image_input_type
-    ImageInputType = VisionLanguageConfig.ImageInputType
+    return seq_data, ImageData(image)
 
-    fake_mm_data: MultiModalData
-    if config_input_type == ImageInputType.PIXEL_VALUES:
-        fake_mm_data = ImagePixelData(values)
-    elif config_input_type == ImageInputType.IMAGE_FEATURES:
-        fake_mm_data = ImageFeatureData(values)
-    else:
-        raise NotImplementedError
-
-    return seq_data, fake_mm_data
-
-
-class ImagePixelData(MultiModalData):
-    """
-    The pixel data of an image. Can be one of:
 
-    - :class:``PIL.Image``: An image object. Requires that a HuggingFace
-      processor is available to the model.
-    - :class:``torch.Tensor``: The raw pixel data which is passed to the model
-      without additional pre-processing.
+class ImageData(MultiModalData):
+    """An :class:``PIL.Image`` image. Requires that a HuggingFace
+    processor is available to the model.
```

Comment:
How do we support the use case when user wants to pass in image embeddings directly? Are we adding another class? (something like `ImageEmbeddingData`)?

Reply:
yes, correct. not in the PR though.